components/org.wso2.carbon.identity.api.server.media.service/org.wso2.carbon.identity.api.server.media.service.v1/src/main/java/org/wso2/carbon/identity/api/server/media/service/v1/core/ServerMediaService.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.api.server.media.service.v1.core;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.cxf.jaxrs.ext.multipart.Attachment;
+import org.wso2.carbon.identity.api.server.common.ContextLoader;
+import org.wso2.carbon.identity.api.server.common.error.APIError;
+import org.wso2.carbon.identity.api.server.common.error.ErrorResponse;
+import org.wso2.carbon.identity.api.server.media.service.common.MediaServiceConstants;
+import org.wso2.carbon.identity.api.server.media.service.common.MediaServiceDataHolder;
+import org.wso2.carbon.identity.api.server.media.service.v1.model.ResourceFilesMetadata;
+import org.wso2.carbon.identity.api.server.media.service.v1.model.ResourceFilesMetadataFileSecurity;
+import org.wso2.carbon.identity.media.StorageSystemManager;
+import org.wso2.carbon.identity.media.exception.StorageSystemException;
+import org.wso2.carbon.identity.media.model.FileSecurity;
+import org.wso2.carbon.identity.media.model.MediaMetadata;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import javax.ws.rs.core.Response;
+
+/**
+ * Perform media service related operations.
+ */
+public class ServerMediaService {
+
+    private static final Log LOG = LogFactory.getLog(ServerMediaService.class);
+
+    /**
+     * Upload media.
+     *
+     * @param filesInputStream      The list of files to be uploaded as input streams.
+     * @param filesDetail           File details of the list of files to be uploaded.
+     * @param resourceFilesMetadata Metadata associated with the file upload.
+     * @param onlyEndUserAccess     If the resource is end-user protected.
+     * @return unique identifier of the uploaded media.
+     */
+    public String uploadMedia(List<InputStream> filesInputStream, List<Attachment> filesDetail,
+                              ResourceFilesMetadata resourceFilesMetadata, boolean onlyEndUserAccess) {
+
+        MediaMetadata mediaMetadata = new MediaMetadata();
+        mediaMetadata.setFileName(filesDetail.get(0).getContentDisposition().getFilename());
```

Comment:
can fileDetail be null? if so handle null check

Reply:
`filesDetail` can't be null because in the API definition it has been made mandatory to have a file (which results in the  `filesInputStream` and `filesDetail` lists during swagger code gen) in the file upload request.